core/src/main/scala/org/apache/spark/scheduler/cluster/SparkDeploySchedulerBackend.scala

```diff
@@ -83,8 +83,16 @@ private[spark] class SparkDeploySchedulerBackend(
       args, sc.executorEnvs, classPathEntries ++ testingClassPath, libraryPathEntries, javaOpts)
     val appUIAddress = sc.ui.map(_.appUIAddress).getOrElse("")
     val coresPerExecutor = conf.getOption("spark.executor.cores").map(_.toInt)
+    if (coresPerExecutor.isDefined && (coresPerExecutor.get < scheduler.CPUS_PER_TASK ||
+      (coresPerExecutor.get % scheduler.CPUS_PER_TASK) != 0)) {
+      throw new SparkException(s"invalid configuration of " +
+        s"spark.executor.cores(${coresPerExecutor.get}) and " +
+        s"spark.task.cpus(${scheduler.CPUS_PER_TASK}}), spark.executor.cores has to be no less " +
+        "than and folds of spark.task.cpus")
```

Comment:
I don't think this error message is super clear, what is meant by "no less than and folds of spark.task.cups"?